tests/unit/model-test.js

```diff
@@ -3207,6 +3210,129 @@ module('unit/model', function(hooks) {
     );
   });
 
+  test('.changedAttributes returns dirty attributes for arrays of primitive values upon updating the array', function(assert) {
+    let model = run(() => {
+      return this.store.push({
+        data: {
+          id: 1,
+          type: 'com.example.bookstore.Book',
+          attributes: {
+            name: 'The Winds of Winter',
+            author: 'George R. R. Martin',
+            chapters: ['Windy eh', 'I guess winter was coming after all'],
+          },
+        },
+      });
+    });
+    assert.ok(!model.get('isDirty'), 'model currently not dirty');
+    assert.equal(
+      model._internalModel.currentState.stateName,
+      'root.loaded.saved',
+      'model.state loaded.saved'
+    );
+
+    // Pushing simple value to array
+    const chapters = model.get('chapters');
+    run(() => {
+      chapters.pushObject('New Chapter');
+    });
+
+    assert.deepEqual(
+      model.changedAttributes(),
+      {
+        chapters: [
+          ['Windy eh', 'I guess winter was coming after all'],
+          ['Windy eh', 'I guess winter was coming after all', 'New Chapter'],
+        ],
+      },
+      '.changedAttributes returns changed arrays'
+    );
+
+    // Pushing simple value to array
+    run(() => {
+      chapters.pushObject({
+        name: 'Windy eh new',
+        number: 1,
+      });
+    });
+
+    assert.ok(chapters.get('lastObject') instanceof MegamorphicModel);
```

Comment:
let's add an assertion that checks the new model has the right attrs

```js
assert.equal(chapters.get('lastObject').get('name'), 'Windy eh new', 'new embedded model has right attrs');
```

Reply:
Added assertions for the attributes. And also fixed test failures locally.